### PR TITLE
feat: added integration tests and other changes

### PR DESCRIPTION
**Pull request by tanweersalah**
# Enhance KymaReActAgent: Add Detailed Instructions, Improve Context Handling, and Add Integration Tests

### New Features

✨ Introduces comprehensive integration tests for `KymaReActAgent` and refactors its prompting system with a structured multi-step ReAct instruction set. Context handling for namespace and cluster overviews is improved to provide more informative fallback messages.

### Changes

* `.vscode/settings.json`: Added `-n auto` to pytest args to enable parallel test execution.
* `src/agents/common/utils.py`: Improved K8s context retrieval — cluster overview now generates descriptive fallback messages when no pods, metrics, or events are found. Namespace overview now verifies namespace existence before fetching events, returning a clear "not found" message if the namespace is missing.
* `src/agents/kyma/prompts.py`: Replaced the monolithic `KYMA_AGENT_PROMPT`/`KYMA_AGENT_INSTRUCTIONS` with two new constants: `REACT_AGENT_PROMPT` (role definition and critical rules) and `REACT_AGENT_INSTRUCTIONS` (a detailed multi-step routing guide covering broad queries, cluster/namespace overviews, specific resource analysis, and concept questions).
* `src/agents/kyma/react_agent.py`: Updated imports and `SYSTEM_PROMPT` to use the new `REACT_AGENT_PROMPT` and `REACT_AGENT_INSTRUCTIONS`.
* `tests/integration/agents/kyma/test_kyma_react_agent.py`: **New file.** Full integration test suite for `KymaReActAgent` covering:
  - Basic invocation and response quality (semantic correctness via `deepeval`)
  - Tool selection accuracy (routing to correct tools per query type, including edge cases for broad/generic queries)
  - `UINavigationContext` handling
  - Multi-turn chat history (follow-ups, answering from history without tool calls)
  - Error handling (K8s failures, long queries, `None` callbacks)
  - Goal accuracy against a live cluster (using `ragas` `SimpleCriteriaScore`)
  - Callback support verification
* `tests/integration/agents/summarization/test_integration_summarization.py`: File renamed (no content changes).
* `tests/unit/initial_questions/test_initial_questions.py`: Added `mock_k8s_client.get_namespace` as an `AsyncMock` to align with the updated namespace-existence check.
* `tests/unit/utils/test_factory.py`: Added `AnthropicModel` test case and fixture; introduced fixed test model name constants to decouple tests from environment-driven settings.
* `tests/unit/utils/test_src_utils.py`: Added `get_namespace` `AsyncMock` to the shared fixture; added a new parametrized test `test_namespace_overview` covering missing-namespace and no-events scenarios.

- [ ] 🔄 Regenerate and Update Summary

<details>
<summary>PR Bot Information</summary>

**Version:** `1.29.2`

- Event Trigger: `issue_comment.edited`
- Correlation ID: `77024f40-878b-11f1-9fe5-47f1ff039978`
</details>

### CODE REVIEW

File: src/agents/common/utils.py [hunk 1]
```diff
@@ -167,17 +167,42 @@ async def get_relevant_context_from_k8s_cluster(message: Message, k8s_client: IK
...
+        if not namespace_exists:
+            context = f"Namespace '{namespace}' was not found in the cluster."
+        else:
+            logger.debug("Fetching all K8s Events with warning type")
+            warning_events = k8s_client.list_k8s_warning_events(namespace=namespace)
+            context = (
+                yaml.dump_all(warning_events)
+                if warning_events
+                else f"Namespace '{namespace}' exists, but no warning or error events were found."
+            )
```
[the remark this answers is not in the document]
Comment (ruanxin) [reply src/agents/common/utils.py:210]: This suggestion makes sense, we should treat “namespace not found” as actual not-found cases (404 / ValueError) and other errors separately in the except Exception block @tanweersalah

File: tests/integration/agents/kyma/test_kyma_agent.py [hunk 2]
```diff
@@ -79,54 +79,54 @@ def kyma_agent(app_models):
...
-            False,
-        ),
-        # Test case for API Rule with wrong access strategy
+        # (
+        #     "Should mention about Joule context in kyma dashboard",
+        #     KymaAgentState(
+        #         agent_messages=[],
+        #         messages=[
+        #             SystemMessage(
+        #                 content="{'resource_kind': 'Function', 'resource_api_version': 'serverless.kyma-project.io/v1alpha2'}"
```
[the remark this answers is not in the document]
Comment (ruanxin) [reply tests/integration/agents/kyma/test_kyma_agent.py:85]: please use pytest skip and give the reason as suggested by the copilot @tanweersalah

File: tests/unit/initial_questions/test_initial_questions.py [hunk 3]
```diff
@@ -76,6 +76,7 @@ def mock_k8s_client():
     mock.list_not_running_pods.return_value = [{KEY: LIST_NOT_RUNNING_PODS}, MOCK_DICT]
     mock.list_nodes_metrics = AsyncMock()
     mock.list_nodes_metrics.return_value = [{KEY: LIST_NODES_METRICS}, MOCK_DICT]
+    mock.get_namespace = AsyncMock(return_value=None)
```
[the remark this answers is not in the document]
Comment (ruanxin) [reply tests/unit/initial_questions/test_initial_questions.py]: Please take this suggestion. @tanweersalah

**Review by ruanxin — APPROVED** [review]
